nixos/modules/tasks/network-namespaces.nix

```diff
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.networking;
+in
+{
+  options = {
+    networking.namespaces = mkOption {
```

Comment:
We could split this into its own PR actually. This would make the current PR pretty straightforward and uncontroversial, and the two options are anyway orthogonal.

WDYT?